### PR TITLE
More AngelScript features

### PR DESCRIPTION
**Pull request by AnotherFoxGuy**
#298 Added ```game.spawnTruck("shoppingcart.truck" ,vector3(512, 0, 512) , vector3(0, 90, 0));```
Spawns a truck and returns the reference of the Beam object

Added ```game.showMessageBox("Im a box","Do you like it?",true,"Yes",false,true,"No");```
Shows a simple MessageBox

### CODE REVIEW

File: doc/angelscript/game.h [hunk 1]
```diff
@@ -288,7 +288,7 @@ class game
 	 * @param instanceName The unique name that you chose when spawning this object
 	 * @param pos The position where the object should be moved to
 	 */
-	void moveObjectVisuals(const Ogre::String& instanceName, const Ogre::Vector3& pos);
+	void moveObjectVisuals(const string instanceName, const vector3 pos);
```
Comment (Hiradur) [inline doc/angelscript/game.h]: Is this necessary? I think the coding guidelines mentioned to use Ogre::String and Ogre::Vector3 but I can't this information anymore.

**Comment by Hiradur** [comment]
This PR would probably break building without AS once again, can you fix it?

**Comment by AnotherFoxGuy** [comment]
I just tried building without AngelScript, it compiles without problems.

File: doc/angelscript/game.h (hunk 1 above)
Comment (AnotherFoxGuy) [inline doc/angelscript/game.h]: The header is only used to generate the [AngelScript documentation](http://developer.rigsofrods.org/angelscript/)

**Comment by Hiradur** [comment]
:+1: from me then

**Pull request merged**